test/unit/org/apache/cassandra/db/SinglePartitionSliceCommandTest.java

```diff
@@ -335,40 +457,49 @@ public void toCQLStringIsSafeToCall() throws IOException
         Assert.assertFalse(ret.isEmpty());
     }
 
-
-    public static List<Unfiltered> getUnfilteredsFromSinglePartition(String q)
+    public static UnfilteredRowIterator getIteratorFromSinglePartition(String q)
     {
         SelectStatement stmt = (SelectStatement) QueryProcessor.parseStatement(q).prepare(ClientState.forInternalCalls());
 
-        List<Unfiltered> unfiltereds = new ArrayList<>();
         SinglePartitionReadQuery.Group<SinglePartitionReadCommand> query = (SinglePartitionReadQuery.Group<SinglePartitionReadCommand>) stmt.getQuery(QueryOptions.DEFAULT, 0);
         Assert.assertEquals(1, query.queries.size());
         SinglePartitionReadCommand command = Iterables.getOnlyElement(query.queries);
         try (ReadExecutionController controller = ReadExecutionController.forCommand(command);
              UnfilteredPartitionIterator partitions = command.executeLocally(controller))
         {
             assert partitions.hasNext();
-            try (UnfilteredRowIterator partition = partitions.next())
-            {
-                while (partition.hasNext())
-                {
-                    Unfiltered next = partition.next();
-                    unfiltereds.add(next);
-                }
-            }
+            UnfilteredRowIterator partition = partitions.next();
             assert !partitions.hasNext();
+            return partition;
+        }
+    }
+
+    public static List<Unfiltered> getUnfilteredsFromSinglePartition(String q)
+    {
+        try (UnfilteredRowIterator partition = getIteratorFromSinglePartition(q))
+        {
+            return toUnfiltereds(partition);
         }
+    }
+
+    private static List<Unfiltered> toUnfiltereds(UnfilteredRowIterator partition)
+    {
+        List<Unfiltered> unfiltereds = new ArrayList<>();
+        while (partition.hasNext())
+            unfiltereds.add(partition.next());
+
```

Comment:
We can use `Lists.newArrayList(partition)`, unless we want to reduce dependency from Guava